tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory_resource>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <utility>
+
+using namespace std;
+
+template <typename Stream>
+struct test_rvalue {
+    void operator()(const string& init_value) {
+        string buffer{init_value};
+        Stream stream{move(buffer)};
+        assert(stream.view() == init_value);

```

Comment:
This is good for testing that all the things compile but is missing memory and allocator consistency checks.